manager/volume.go

```diff
@@ -483,11 +484,8 @@ func (m *VolumeManager) Activate(volumeName string, frontend string) (v *longhor
 		engine = e
 	}
 
-	if v.Status.LastBackup != engine.Status.LastRestoredBackup || engine.Spec.RequestedBackupRestore != engine.Status.LastRestoredBackup {
-		logrus.Infof("Standby volume %v will be activated after finishing restoration, "+
-			"backup volume's latest backup: %v, "+
-			"engine requested backup restore: %v, engine last restored backup: %v",
-			v.Name, v.Status.LastBackup, engine.Spec.RequestedBackupRestore, engine.Status.LastRestoredBackup)
+	if done, err := m.syncBackupAndRestoreWithLastBackup(v, engine); !done {
```

Comment:
`Activate` is used by the REST API, so this PR will make this function become from blocking/sync from non-blocking/async. This could cause some unexpected user experiences. and probably still not working if the request has not been processed completely because it's synchronous.

I would suggest checking the below conditions before attaching the activating DR volume. 
- The backup volume sync is updated
- The restored backup of the engine is updated

[/controller/volume_controller.go#L1637-L1642](https://github.com/longhorn/longhorn-manager/blob/788bb3e7c061baa90e7abc99bdb1f860b5a3efe4/controller/volume_controller.go#L1637-L1642)
```go
		e.Spec.NodeID = v.Status.CurrentNodeID
		e.Spec.ReplicaAddressMap = replicaAddressMap
		e.Spec.DesireState = longhorn.InstanceStateRunning
		// The volume may be activated
		e.Spec.DisableFrontend = v.Status.FrontendDisabled
		e.Spec.Frontend = v.Spec.Frontend
```

The thing you need to do in `Activate` is just update `backupVolume.Spec.SyncRequestedAt` as your current PR to ensure the backup volume will be reconciled to update its latest backup.

Reply:
I have updated to the following description:
 - `Activate` RESTful API: trigger backup volume to update to get the latest backup name
 - make sure restoring with the latest backup before detach volume when activate DR volume
   - Since the DR volume will be detached after activation, and it will also make the engine stopped. Therefore, I add the check when detach the volume.